tasks/swarm.yml

```diff
@@ -1,3 +1,52 @@
 ---
-# TODO
-# https://www.digitalocean.com/community/tutorials/how-to-create-a-cluster-of-docker-containers-with-docker-swarm-and-digitalocean-on-ubuntu-16-04
+- name: Checking Docker Swarm mode status
+  command: "docker info"
+  register: "docker_info"
+  changed_when: false
+
+- name: Docker Swarm initalization
+  command: >
+           docker swarm init
+           --listen-addr {{ hostvars[inventory_hostname]['ansible_' + docker_swarm_interface]['ipv4']['address'] }}:{{ docker_swarm_port }}
+           --advertise-addr {{ hostvars[inventory_hostname]['ansible_' + docker_swarm_interface]['ipv4']['address'] }}
+  when: '"Swarm: inactive" in docker_info.stdout'
+  run_once: yes
+  delegate_to: "{{ docker_swarm_primary_manager }}"
+
+- block:
+  - name: Capturing Docker Swarm worker join token
+    command: "docker swarm join-token -q worker"
+    changed_when: false
+    register: "docker_swarm_worker_token"
+
+  - name: Capturing Docker Swarm manager join token
+    command: "docker swarm join-token -q manager"
+    changed_when: false
+    register: "docker_swarm_manager_token"
+  run_once: yes
+  delegate_to: "{{ docker_swarm_primary_manager }}"
+
+- name: Joining additional Docker Swarm managers to cluster
+  command: >
+           docker swarm join
+           --listen-addr {{ hostvars[inventory_hostname]['ansible_' + docker_swarm_interface]['ipv4']['address'] }}:{{ docker_swarm_port }}
+           --advertise-addr {{ hostvars[inventory_hostname]['ansible_' + docker_swarm_interface]['ipv4']['address'] }}
+           --token {{ docker_swarm_manager_token.stdout }}
+           {{ hostvars[docker_swarm_primary_manager]['ansible_' + docker_swarm_interface]['ipv4']['address'] }}
+  when:
+    - inventory_hostname != docker_swarm_primary_manager
+    - inventory_hostname not in groups['docker-swarm-workers']
```

Comment:
It would be good to provide group as a variable not as a static name 'docker-swarm-workers'.
Like this: https://github.com/SoInteractive/docker/blob/master/defaults/main.yml#L2